src/lib.rs

```diff
@@ -569,6 +613,8 @@ pub struct PerCrateAnalysis {
     ref_spans: HashMap<Id, Vec<Span>>,
     globs: HashMap<Span, Glob>,
     impls: HashMap<Id, Vec<Span>>,
+    per_fn_borrows: HashMap<Id, BorrowData>,
+    fn_span_lookup_tree: BTreeMap<SearchSpan, Id>,
```

Comment:
I had an alternative design where rather than storing `SearchSpan` here, I stored a type called `InTreeSearchSpan`, which was just a wrapper around `SearchSpan(some_span, SpanKind::InTree)`. Though in that case, I had to implement `Borrow<SearchSpan>` as well as all the `Eq` and `Ord` traits. It's safer type-wise since one wouldn't be able to accidentally store the `SpanKind`'s `Start` and `End` values like you can now. However, I wasn't sure if it's worth the extra maintenance cost since the usage is so simple and easy to verify for correctness. Let me know if you'd like me to add it back in (assuming this `SearchSpan` and `BTreeMap` combination is something you're ok with).